test/integration/resourcemanager/nodeagentauthorizer/nodeagentauthorizer_suite_test.go

```diff
@@ -0,0 +1,244 @@
+}
+
+const testID = "nodeagentauthorizer-webhook-test"
+
+var (
+	ctx = context.Background()
+	log logr.Logger
+
+	sourceRestConfig *rest.Config
+	sourceEnv        *envtest.Environment
+	sourceClient     client.Client
+
+	targetRestConfig          *rest.Config
+	targetRestConfigNodeAgent *rest.Config
+	targetEnv                 *envtest.Environment
+	targetClient              client.Client
+	targetClientNodeAgent     client.Client
+
+	testRunID     string
+	testNamespace *corev1.Namespace
+
+	machineName       string
+	userNameNodeAgent string
+)
+
+var _ = BeforeSuite(func() {
+	logf.SetLogger(logger.MustNewZapLogger(logger.DebugLevel, logger.FormatJSON, zap.WriteTo(GinkgoWriter)))
+	log = logf.Log.WithName(testID)
+	testRunID = utils.ComputeSHA256Hex([]byte(uuid.NewUUID()))[:16]
+
+	// determine a unique namespace name for test environment
+	testNamespaceName := testID + "-" + testRunID[:8]
+
+	By("Start source environment")
+	sourceEnv = &envtest.Environment{
+		CRDInstallOptions: envtest.CRDInstallOptions{
+			Paths: []string{
+				filepath.Join("..", "..", "..", "..", "example", "seed-crds", "10-crd-machine.sapcloud.io_machines.yaml"),
+			},
+		},
+		ErrorIfCRDPathMissing: true,
+	}
+
+	var err error
+	sourceRestConfig, err = sourceEnv.Start()
+	Expect(err).NotTo(HaveOccurred())
+	Expect(sourceRestConfig).NotTo(BeNil())
+
+	DeferCleanup(func() {
+		By("Stop source environment")
+		Expect(sourceEnv.Stop()).To(Succeed())
+	})
+
+	By("Create source client")
+	sourceClient, err = client.New(sourceRestConfig, client.Options{Scheme: resourcemanagerclient.CombinedScheme})
+	Expect(err).NotTo(HaveOccurred())
+
+	By("Create test Namespace")
+	testNamespace = &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			// create dedicated namespace for each test run, so that we can run multiple tests concurrently for stress tests
+			Name: testNamespaceName,
+		},
+	}
+	Expect(sourceClient.Create(ctx, testNamespace)).To(Succeed())
+	log.Info("Created test Namespace in source cluster", "namespaceName", testNamespace.Name)
+
+	DeferCleanup(func() {
+		By("Delete test Namespace from source cluster")
+		Expect(sourceClient.Delete(ctx, testNamespace)).To(Or(Succeed(), BeNotFoundError()))
+	})
+
+	By("Create kubeconfig file for source cluster")
+	kubeconfigFileName, err := createKubeconfigFileForAuthorizationWebhook(sourceEnv.WebhookInstallOptions)
+	Expect(err).ToNot(HaveOccurred())
+	DeferCleanup(func() {
+		By("Delete kubeconfig file for source cluster")
+		Expect(os.Remove(kubeconfigFileName)).To(Succeed())
+	})
+
+	By("Start target environment")
+	Expect(framework.FileExists(kubeconfigFileName)).To(BeTrue())
+	testAPIServer := &envtest.APIServer{}
+	testAPIServer.Configure().
+		Set("authorization-mode", "RBAC", "Webhook").
+		Set("authorization-webhook-config-file", kubeconfigFileName).
+		Set("authorization-webhook-cache-authorized-ttl", "0").
+		Set("authorization-webhook-cache-unauthorized-ttl", "0")
```

Comment:
Maybe let's use structured authz instead? We already use 1.30 in our envtests.

I'm still no fan of running two envtests. I don't think we are doing this anywhere else in our integration tests. IIUC, your concern was that this authz webhook must be reachable for the KAPI to start. With structured authz, you can define matching conditions, i.e., the webhook gets only consulted for relevant API calls. This way, the webhook server can also start later.

Reply:
There is another reason for running two envtest requirements. We need the port and the CA of the envtest environment for the kubeconfig of authorization webhook. The kubeconfig must be available when the envtest environment is starting, but port and CA are filled after envtest was started (unless you want to create them manually).

Reply:
In the scope of the integration test, I think we can ignore the CA and skip the TLS verification. As for the port: We have a similar "challenge" when starting the GAPI in the env tests. Here, we fetch the port upfront and then configure it accordingly: https://github.com/gardener/gardener/blob/3c1157fe3a713647e3ceba4555c5619014f2cf0f/test/envtest/apiserver.go#L232-L237
So, we could do something similar with just a few lines: 

```patch
        log = logf.Log.WithName(testID)

        By("Start test environment")
+       securePort, _, err := port.SuggestPort("")
+       Expect(err).NotTo(HaveOccurred())
+
        testEnv = &envtest.Environment{
+               ControlPlane: envtest.ControlPlane{
+                       APIServer: &envtest.APIServer{
+                               SecureServing: envtest.SecureServing{
+                                       ListenAddr: envtest.ListenAddr{
+                                               Port: strconv.Itoa(securePort),
+                                       },
+                               },
+                       },
+               },
```

WDYT?

Reply:
It works. We don't even have to configure structured authorization 🙂

Reply:
I played around with the operator and enabled `SeedAuthorizer` right from the start ([ref](https://github.com/oliver-goetz/gardener/commit/09de13e76ebf56d7ee76353d9703114329511327)). In this setup, it still works to create a Garden which makes sense since RBAC still has first priority. Did I miss something or can we reconsider the requirement of deploying the apiservers twice?
This is a bit off-topic, but it came to my mind because the envtest apiserver was coming up when its authorization webhook was unreachable.